### PR TITLE
Add vote

### PR DESCRIPTION
**Pull request by kpat108**
Please review.  Would the votes need to have tests as well?

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -4,6 +4,11 @@ class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
 
 
+  def render_not_found(status=:not_found)
+    render text: "#{status.to_s.titleize} :(", status: status
+  end
```
Comment (kenmazaika) [inline app/controllers/application_controller.rb]: I really like this!

File: app/controllers/shares_controller.rb [hunk 2]
```diff
@@ -8,17 +10,62 @@ def new
...
+		if @share.user != current_user
+			return render text: 'Not Allowed', status: :forbidden
+		end
+
+		@share.update_attributes(share_params)
+		if @share.valid?
+			redirect_to shares_path
+		else
+			render :new, status: :unprocessable_entity
+    end
```
Comment (kenmazaika) [inline app/controllers/shares_controller.rb]: Indentation looks off here.

File: app/controllers/votes_controller.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+class VotesController < ApplicationController
+
+  def index
```
Comment (kenmazaika) [inline app/controllers/votes_controller.rb]: Is this actually used?

File: spec/factories.rb [hunk 4]
```diff
@@ -7,4 +7,10 @@
 		password_confirmation "secretPassword"
 
 	end
+
+  factory :share do 
+    body "Hack Attack!"
```
Comment (kenmazaika) [inline spec/factories.rb]: Hack Attack!

**Comment by kenmazaika** [comment]
Nice job.  Thumbs up for adding the infrastructure for testing too!

**Comment by kpat108** [comment]
I am not sure why the indentation looks off.  It is looking normal on sublime. 
screenshot_1

The votes controller is not being used.  I was trying to follow the MVC idea that we have been using throughout the different projects.  I also added it in case any tests need to be written for creating votes.  Would you like me to delete it?

**Comment by kenmazaika** [comment]
I think we should remove the controller until it's actually used.  

The indentation problem is tabs vs. spaces, which is pretty invisible in sublime text but becomes apparent on GitHub:
http://haacked.com/archive/2013/09/17/popular-code-conventions-on-github.aspx/

**Comment by kpat108** [comment]
Wow, I did not know about that.  Thank you.

**Comment by kenmazaika** [comment]
No problem!

**Comment by kpat108** [comment]
I have made the recommended changes

**Comment by kenmazaika** [comment]
Looks great.  Nice job reworking the indentation!

**Pull request merged**